### PR TITLE
New Model for Employee, Update old project model, etc. 

### PR DESCRIPTION
**Pull request by TarikVelic**
--------------------------------------------------------------------
 MAIN
 * Added new Model for Employee
 * Updated the old project model so that it looks for an employee instead of a user
 * Deleted some things from the user model, service and etc that we dont need when employee model is there --------------------------------------------------------------------
 OTHER
 *
--------------------------------------------------------------------

### CODE REVIEW

File: controllers/employee.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    res.status(201).json(employee);
+  } catch (error) {
+    res.status(500).json({ error: errors.FAILED_TO_CREATE_EMPLOYEE });
+  }
+};
+
+exports.getEmployeeById = async (req, res) => {
+  try {
+    const employee = await employeeService.getEmployeeById(req.params.id);
+    if (!employee) {
```
Comment (becarselman) [inline controllers/employee.js]: You will do this in service. You can check this and throw error for this case

File: db/models/employee.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    type: String,
+    required: true,
+  },
+  profileImage: {
+    type: String,
+    required: false, 
+  },
+  department: {
+    type: String,
+    enum: ['Management', 'Administration', 'Design', 'Development'],
```
Comment (becarselman) [inline db/models/employee.js]: can we specify this enum variables in some utils file for example utils/enums where will be all enums

File: controllers/employee.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    } else {
+      res.status(500).json({ error: errors.FAILED_TO_GET_EMPLOYEE });
+    }
+  }
+};
+
+exports.getAllEmployees = async (req, res) => {
+  try {
+    const employees = await employeeService.getAllEmployees();
+    res.json(employees);
```
Comment (becarselman) [inline controllers/employee.js]: You are missing status

File: controllers/employee.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    res.status(201).json(employee);
+  } catch (error) {
+    res.status(500).json({ error: errors.FAILED_TO_CREATE_EMPLOYEE });
+  }
+};
+
+exports.getEmployeeById = async (req, res) => {
+  try {
+    const employee = await employeeService.getEmployeeById(req.params.id);
+    res.json(employee);
```
Comment (becarselman) [inline controllers/employee.js]: you are missing status code

File: controllers/employee.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
+const employeeService = require('../services/employee');
+const errors = require('../configuration/errors');
+
+exports.createEmployee = async (req, res) => {
+  try {
+    const employee = await employeeService.createEmployee(req.body);
+    res.status(201).json(employee);
```
Comment (becarselman) [inline controllers/employee.js]: Also you are missing return here. You need to return this and on all places similar to this

File: controllers/employee.js [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+    res.json(employees);
+  } catch (error) {
+    res.status(500).json({ error: errors.FAILED_TO_GET_EMPLOYEE });
+  }
+};
+
+exports.updateEmployee = async (req, res) => {
+  try {
+    const employee = await employeeService.updateEmployee(req.params.id, req.body);
+    if (!employee) {
```
Comment (becarselman) [inline controllers/employee.js]: If I leave comment on one place please correct on all places where you did same mistake

File: repositories/employee.js
Comment (becarselman) [inline repositories/employee.js:1]: Where is pagination here.

File: controllers/projects.js [hunk 7]
```diff
@@ -1,9 +1,25 @@
...
   async createProject(req, res, next) {
     try {
-      const project = await projectService.createProject(req.body);
+      const { developers, ...projectData } = req.body;
+      const projectDevelopers = [];
+
+      for (const developer of developers) {
+        const employee = await employeeService.getEmployeeById(developer.employee);
+        if (!employee) {
+          return res.status(400).json({ error: 'Employee not found.' });
```
Comment (becarselman) [inline controllers/projects.js]: you can reuse error message not found

File: controllers/projects.js [hunk 8]
```diff
@@ -1,9 +1,25 @@
...
+const employeeService = require("../services/employee");
 
 module.exports = {
   async createProject(req, res, next) {
     try {
-      const project = await projectService.createProject(req.body);
+      const { developers, ...projectData } = req.body;
+      const projectDevelopers = [];
+
+      for (const developer of developers) {
```
Comment (becarselman) [inline controllers/projects.js]: this need to be done in service

File: repositories/employee.js [hunk 9]
```diff
@@ -0,0 +1,63 @@
+const Employee = require('../db/models/employee');
+
+class EmployeeRepository {
+  async createEmployee(employeeData) {
+    try {
+      const employee = new Employee(employeeData);
```
Comment (becarselman) [inline repositories/employee.js]: use orm

File: services/employee.js [hunk 10]
```diff
@@ -0,0 +1,53 @@
+const employeeRepository = require('../repositories/employee');
+
+class EmployeeService {
```
Comment (becarselman) [inline services/employee.js]: why you start using class if we use functions before. Please stay consistent

**Review by becarselman — CHANGES_REQUESTED** [review]

**Review by becarselman — APPROVED** [review]
I will approve this, But I have additional questions please ping me over discord

**Review by melcidar — APPROVED** [review]

File: services/employee.js [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+      return employee;
+    } catch (error) {
+      throw new Error(errors.FAILED_TO_GET_EMPLOYEE);
+    }
+  },
+
+  async getAllEmployees() {
+    try {
+      const employees = await employeeRepository.getAllEmployees();
+      const simplifiedEmployees = employees.map((employee) => {
```
Comment (becarselman) [inline services/employee.js]: this is not how we do this. Please use orm and do it in repository. This approach is same as before. We need to do this on db call and not here. https://stackoverflow.com/questions/67809187/mongoose-how-to-return-only-specific-field

**Review by becarselman — CHANGES_REQUESTED** [review]

File: services/employee.js (hunk 11 above)
Comment (TarikVelic) [reply services/employee.js]: fixed

**Review by becarselman — APPROVED** [review]

**Pull request merged**